webservices/ruby/style_guide.md

```diff
@@ -0,0 +1,492 @@
+* **[Use a trailing comma after the last item in a multi-line list](#use-a-trailing-comma-after-the-last-item-in-a-multi-line-list)**
+* **[Avoid explicit return statements](#avoid-explicit-return-statements)**
+* **[Avoid multiple assignments per line](#avoid-multiple-assignments-per-line)**
+* **[Prefer && and || over AND and OR](#prefer--and--over-and-and-or)**
+* **[Prefer ! over not](Prefer--over-not)**
+* **[Use %{} for single-line strings needing interpolation and double-quotes](use--for-single-line-strings-needing-interpolation-and-double-quotes)**
+* **[Avoid using semicolons](#avoid-using-semicolons)**
+* **[Avoid organizational comments](#avoid-organizational-comments)**
+* **[Use CamelCase for classes and modules, snake_case for variables, methods, and file names, SCREAMING_SNAKE_CASE for constants.](#use-camelcase-for-classes-and-modules-snake_case-for-variables-methods-and-file-names-screaming_snake_case-for-constants)**
+
+## [Conditionals](#conditionals)
+* **[Avoid lines that end with conditionals](#avoid-lines-that-end-with-conditionals)**
+* **[Avoid ternary operators](#avoid-ternary-operators)**
+* **[Prefer if over unless](#prefer-if-over-unless)**
+
+## [Blocks](#blocks)
+* **[Prefer &:method_name for simple method calls](#prefer-method_name-for-simple-method-calls)**
+* **[Use {...} for single-line blocks. Use do..end for multi-line blocks](#use--for-single-line-blocks-use-doend-for-multi-line-blocks)**
+* **[Prefer detect, select, map, and reduce for iteration](#prefer-detect-select-map-and-reduce-for-iteration)**
+
+## [Classes & Methods](#classes--methods)
+* **[Use ? suffix for predicate methods, and avoid prefixing auxiliary verbs](#use--suffix-for-predicate-methods-and-avoid-prefixing-auxiliary-verbs)**
+* **[Use def with parentheses when there are arguments](#use-def-with-parentheses-when-there-are-arguments)**
+* **[Don't use spaces after required keyword arguments](#dont-use-spaces-after-required-keyword-arguments)**
+* **[Avoid bang method names](#avoid-bang-method-names)**
+* **[Don't use self explicitly except for class methods and assignments](#dont-use-self-explicitly-except-for-class-methods-and-assignments)**
+* **[Prefer shorthand class and module definitions](#prefer-shorthand-class-and-module-definitions)**
+* **[Order class methods above instance methods](#order-class-methods-above-instance-methods)**
+
+## General
+
+### Prefer double quotes for strings.
+  ```ruby
+    # bad
+      'Kaladin Stormblessed'
+
+    # good
+      "Kaladin Stormblessed"
+  ```
+
+### Use spaces around curly braces
+  ```ruby
+    # bad
+    {name: "Kelsier", title: "Survivor of Hathsin"}
+
+    # good
+    { name: "Kelsier", title: "Survivor of Hathsin" }
+  ```
+### Don't use spaces around square brackets
+  ```ruby
+    # bad
+    [ "Frodo, Gandalf, Aragorn, Legolas, Gimli, Sam, Merry, Pippin" ]
+
+    # good
+    ["Frodo, Gandalf, Aragorn, Legolas, Gimli, Sam, Merry, Pippin"]
```

Comment:
not wrong necessarily but these are just 1 string with commas in them, rather than a list of strings. Not sure if that was intentional or not

Reply:
I agree. Also, this is confusing.  You'd hopefully never see a list of strings separated with commas and spaces inside a string; that's what the computer produces.  As soon as I saw this, I was thinking about %w(Frodo Gandalf Aragorn Legolas etc) which would produce the list of the names properly quoted.